### ISSUE TITLE
OptionalJsonColumn ignoring not parsable values

### ISSUE THREAD

**Issue by aludwiko** [outworkers/phantom#813]
`OptionalJsonColumn` is using a generic strategy for parsing values:
```
  class OptionalCol[RR](
    implicit ev: Primitive[RR],
    evOpt: Primitive[Option[RR]]
  ) extends Col[Option[RR]] {
    override def parse(r: Row): Try[Option[RR]] = ev.fromRow(name, r) match {
      case Success(value) => Success(Some(value))
      case Failure(_) => Success(None)
    }
  }
```
which in case of not parsable value will swallow the exception and return `None`. 
`None` vs. parsing exception is something completly different from business point of view.

**Comment by alexflav23** [comment]
Hi @aludwiko This is a fair point, but you shouldn't really be inserting bad values in Cassandra, it's important to guarantee the store and retrieve function is a bijection, meaning you are not losing any valuable information by going "in and out" of Cassandra with the same bit of data.

You are pushing an application level concern to the library in that sense, `Option` columns were never designed to cope with parsing errors, which is exactly why they swallow them up. They are a DSL nicety around Cassandra `null`, not around inserting invalid JSON.

All that being said, we want to help here, so what do you feel would be the useful API?

**Comment by aludwiko** [comment]
@alexflav23 you are right about the bijections function, but I hit this problem when I was replacing existing Cassandra access library with Phantom. All data was added by a completely different tool. That's why bijection is not possible in my case, at least until I'm able to read correctly all jsons. 

My workaround is quite simple:
```
class MyOptionalJsonColumn[RR](implicit ev: Primitive[RR], optEv: Primitive[Option[RR]]) extends OptionalJsonColumn[RR]() {
    override def parse(r: Row): Try[Option[RR]] = ev.fromRow(name, r) match {
      case Success(value) => Success(Some(value))
      case Failure(exception) => {
        if (exception.getMessage.matches("Column .* is null")) {
          Success(None)
        } else {
          Failure(exception)
        }
      }
    }
  }
```
Matching exception by the message is not the best idea, but currently, I think it is not possible to match it more type safety, so maybe providing a better way to match the exception could be a good start.

**Comment by alexflav23** [comment]
@aludwiko Fixed in 2.21.0, on Maven Central in a few hours.
